syntax = "proto3";

package transaction_verification;

message TransactionVerificationRequest {
  User user = 1;
  CreditCard creditCard = 2;
  Item item = 3;
  VectorClock vectorClock = 4;
}

message User {
    string name = 1;
    string contact = 2;
}

message CreditCard {
    string number = 1;
    string expirationDate = 2;
    string cvv = 3;
}

message Item {
    string name = 1;
    int32 quantity = 2;
}

message VectorClock {
  repeated int32 vcArray = 1;
  double timestamp = 2;
}

message TransactionVerificationResponse {
  bool is_valid = 1;
}

service TransactionVerificationService {
  rpc VerifyTransaction(TransactionVerificationRequest) returns (TransactionVerificationResponse);
}
